.github/copilot-instructions.md

```diff
@@ -0,0 +1,100 @@
+
+### External Resources
+
+When the developer provides HTTPS links in conversation:
+
+- You are permitted and encouraged to fetch content from HTTPS URLs using the `fetch_webpage` tool
+- This applies to documentation, GitHub issues, pull requests, specifications, RFCs, and other web-accessible resources
+- Use the fetched content to provide accurate, up-to-date information in your responses
+- If the link is not accessible or the content is unclear, report this explicitly
+- Always verify that fetched information is relevant to the developer's question before incorporating it into your response
+
+## Workspace Environment Setup
+
+### setup-env.sh Integration
+
+When working in this workspace, always source `setup-env.sh` before executing commands that depend on the build environment:
+
+- **Repository version**: `srcs/phlex/scripts/setup-env.sh` - Multi-mode environment setup for developers
+  - Supports standalone repository, multi-project workspace, Spack, and system packages
+  - Gracefully degrades when optional dependencies (e.g., Spack) are unavailable
+  - See `srcs/phlex/scripts/README.md` for detailed documentation
+- **Workspace version**: `setup-env.sh` (workspace root) - Optional workspace-specific configuration
+  - May exist in multi-project workspace setups
+  - Can supplement or override repository setup-env.sh
+
+Command execution guidelines:
+
+- Use `. ./setup-env.sh && <command>` for terminal commands in workspaces with root-level setup-env.sh
+- Use `. srcs/phlex/scripts/setup-env.sh && <command>` when working in standalone repository
+- Ensure VS Code tasks include appropriate `source` command in their definitions
+- Terminal sessions should source the setup script to access build tools (gcc, cmake, ninja, etc.)
+- VS Code settings should use absolute paths or `${workspaceFolder}/local` rather than environment variables for IntelliSense configuration
+- Always ensure that the terminal's current working directory is appropriate to the command being issued
+
+### Source Directory Symbolic Links
+
+If the workspace root contains a `srcs/` directory, it may contain symbolic links (e.g., `srcs/cetmodules` links to external repositories):
+
+- Always use `find -L` or `find -follow` when searching for files in source directories
+- This ensures `find` follows symbolic links and discovers all actual source files
+- Without this flag, `find` will skip linked directories and miss significant portions of the codebase
+
+## Text Formatting Standards
+
+**CRITICAL: Apply to ALL files you create or edit (bash scripts, Python, C++, YAML, Markdown, etc.)**
+
+- All text files must end with exactly one newline character and no trailing blank lines
+- **Never add trailing whitespace on any line** (spaces or tabs at end of lines)
+- This includes blank lines - they should contain only the newline character, no spaces or tabs
+- Exception: Markdown two-space line breaks (avoid; use proper paragraph breaks instead)
```

Comment:
If there is a better alternative, why allow the lesser alternative?

Reply:
It probably does not matter much (i.e. your preference wins), I would try to phrase it as follow though:
```suggestion
- Exception: Markdown use proper paragraph breaks unless the existing text already uses two-space line breaks.
```